scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -1574,3 +1574,26 @@ def func(x):
         # after the '=', so if necessary, the text could be reduced to, say,
         # "MAXCV = 0.".
         assert "MAXCV = 0.414" in result.message
+
+    def test_strategy_func(self):
+
+        init = [(-8, 0.5)] * 5
+        bounds = [(-8.0,8.0)] * 2
+
+        def func(x):
+            return abs(x[0]) * abs(x[1])
+            
+        def custom_strategy_fn(candidate, population, rng):
+            assert(np.array_equal(candidate, [-8, 0.5]) or np.array_equal(candidate, [2.0, -1.0]))
+            return [2.0, -1.0]	# Ignore the candidate; just be recognizable.
+
+        result = differential_evolution(func,
+                                        init=init,
+                                        bounds=bounds,
+                                        popsize=5,
+                                        maxiter=2,
+                                        polish=False,
+                                        strategy_func=custom_strategy_fn)
+        assert(result.success is True)
+        assert np.array_equal(result.x, [2.0, -1.0]) 
```

Comment:
Please add tests for when the strategy_func does not return an array of the correct size. You need to test:

```
with raises(RuntimeError, match="strategy_func")
```

Reply:
In offline testing can you see what happens when you supply a function with the wrong signature. What does the error look like? Should we handle that `TypeError` ourselves?

Reply:
> Please add tests for when the strategy_func does not return an array of the correct size. You need to test:
> 
> ```
> with raises(RuntimeError, match="strategy_func")
> ```

Will hold off on this until the above (https://github.com/scipy/scipy/pull/19196#discussion_r1318221398) is resolved.

> In offline testing can you see what happens when you supply a function with the wrong signature. What does the error 
> look like? Should we handle that TypeError ourselves?

Will hold off on this until this (https://github.com/scipy/scipy/pull/19196#discussion_r1317930374) is revolved.

Reply:
I just resolved it manually.

Reply:
> I just resolved it manually.

By all means, commit anything you want. And if anything looks to be too big of a change, feel free to just task it back to me.  :)